OJ-1838: add local dev for developing local

jkunle opened the pull request:
## Proposed changes

Add `localdev` environment i.e.

```
  Environment:
    Description: "The environment type"
    Type: "String"
    AllowedValues:
      - "dev"
      - "localdev"
      - "build"
      - "staging"
      - "integration"
      - "production"
    ConstraintDescription: must be dev, localdev, build, staging, integration or production
```
`localdev` is solely used for local development scenario only you have to adjust `cris-local-dev` in `ipv-config` stubs directory i.e.

    tokenUrl: https://xxxx.execute-api.eu-west-2.amazonaws.com/localdev/token
    credentialUrl: https://xxxx.execute-api.eu-west-2.amazonaws.com/localdev/credential/issue 

And also the `.env` of the toy-front endpoint would have to point to a private api 

API_BASE_URL=https://xxxx.execute-api.eu-west-2.amazonaws.com/localdev/

`deploy.sh` is only used locally to deploy stack, should also point to a deployed `common-cri-api-local`
which has been modified to have it's redirect_uri pointing to localhost:8085
also has hardcoded `Environment=localdev`
.
### What changed
https://github.com/govuk-one-login/ipv-cri-toy-api/pull/48 fixed core stub dev
however, we've lost the ability to do local development with toy-api

### Why did it change

When experimenting on toy-api, is preferrable to be able to run the toy front end and stub locally and hit the api

### Issue tracking

- [OJ-1838](https://govukverify.atlassian.net/browse/OJ-1838)

## Checklists

### Environment variables or secrets

<!-- Delete if changes DO include new environment variables or secrets -->

- [ ] No environment variables or secrets were added or changed

<!-- Delete if changes DO NOT include new environment variables or secrets -->

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks


[OJ-1838]: https://govukverify.atlassian.net/browse/OJ-1838?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jkunle committed:
OJ-1838: Added is isLocalDevEnvironment

- is LocalDevEnvironment added for local development since Dev is now
  in the VPC, a specific environment = localdev is now neededed for local development
- Removed isNotDevEnvironment, replaced with isNotDevLikeEnvironment
  effectively this would exclude resource creation for both isLocalDevEnvironment
  and isDevEnvironment
- Added DevOnlyToyApi public api

jkunle committed:
OJ-1838: local development doesn't need api-key

jkunle committed:
OJ-1838: point deploy.sh to common-cri-api-local

common-cri-api-local has redirect_url value pointing to localhost
i.e localhost:8085

Connor-Johnson-DCO commented on infrastructure/public-api.yaml:
```diff
@@ -87,7 +87,11 @@ paths:
-        - api_key: []
+        - api_key:
+            Fn::If:
+              - IsLocalDevEnvironment
```
I don't think this works as you expect. I had to remove this to get api keys working in the other environments.

jkunle replied:
So the issue here is that for some weird reason, cloud formation does deploy the `public-api`, on subsequent deployment of the stack. This only works first time around (which is fine).

jkunle committed:
OJ-1838: added flag to always deploy public-api

This always Api-gateway to pick changes to the public-api
i.e. to use or not to use api-keys in dev scenario
